### PR TITLE
Update Software authentication docs

### PR DESCRIPTION
**Pull request by jwitz**
This PR:
- Adds tabs for IdP integration types
- Does a general, but not comprehensive language and style cleanup in "Integrate an Auth System"
- Moves "Integrate an Auth System" to a more logical place in the sidebar.

### CODE REVIEW

File: software/integrate-auth-system.md [hunk 1]
```diff
@@ -225,76 +167,144 @@ astronomer:
...
+#### Step 1: Create a user pool in Cognito
 
-You'll need an Auth0 account in order to set up connections with the identity management provider of your choice. [Sign up for an Auth0 account](https://auth0.com/signup) if you need to.
+Start by creating a user pool in Cognito. You can either review the default settings or step through them to customize.
 
-#### Create Auth0 tenant domain
+Make sure that you create an `App client`, which is the OpenID client configuration that we will use to authenticate against. You do not need to generate a client secret, as Astronomer is a public client that uses implicit flow.
 
-When you log into Auth0 you'll be prompted to create a tenant domain. You can use the default or your own unique `tenant-name`. Your full tenant domain will look something like `astronomer.auth0.com`.
+Once the pool and app client are created, open `App integration` >`App client settings` and configure the following settings:
```
Comment (lzdanski) [inline software/integrate-auth-system.md:232]:
```suggestion
After Auth0 creates the pool and app client, open `App integration` >`App client settings` and configure the following settings:
```

File: software/integrate-auth-system.md [hunk 2]
```diff
@@ -225,76 +167,144 @@ astronomer:
...
 
-> **Note:** Your full tenant domain may differ if you've created it outside of the United States.
+- Select an identity provider to use (either the built-in cognito user pool or a federated identity provider).
+- Set the callback URL parameter to `https://houston.BASEDOMAIN/v1/oauth/redirect/`.
+- Enable `Implicit grant` in `Allowed OAuth Flows`. Leave the other settings disabled.
+- Enable `email`, `openid`, and `profile` in `Allowed OAuth Scopes`.
 
-#### Create a connection between Auth0 and your identity management provider
+Then switch over to the `Domain name` tab and select a unique domain name to use for your hosted Cognito components.
```
Comment (lzdanski) [inline software/integrate-auth-system.md:239]:
```suggestion
Then, switch to the **Domain name** tab and select a unique domain name to use for your hosted Cognito components.
```

File: software/integrate-auth-system.md [hunk 3]
```diff
@@ -225,76 +167,144 @@ astronomer:
...
+              response_type: token
+```
 
-**Enable / disable desired connections:**
+Your Cognito pool ID can be found in the `General settings` tab of the Cognito portal. Your client ID is found in the `App clients` tab.
 
-* Navigate to `https://manage.auth0.com/dashboard/us/<tenant-name>/applications`.
-* Under `Applications`, select `Default App`.
-* Click the `Connections` tab. You should see your connection created in Step 3 listed here. Enable your new connection, and disable any connections that you won't be using.
+Once you've saved your `config.yaml` file with these values, push it to your platform. See [Apply a config change](apply-platform-config.md).
```
Comment (lzdanski) [inline software/integrate-auth-system.md:261]:
```suggestion
After you save your `config.yaml` file with these values, push it to your platform. See [Apply a config change](apply-platform-config.md).
```
Is *push* the correct verb? Is it a sync, a command?

File: software/integrate-auth-system.md [hunk 4]
```diff
@@ -225,76 +167,144 @@ astronomer:
...
-* Navigate to `https://manage.auth0.com/dashboard/us/<tenant-name>/applications`.
-* Under `Applications`, select `Default App`.
-* Click the `Settings` tab.
-* Under `Allowed Callback URLs`, add `https://houston.<your-astronomer-base-domain>/v1/oauth/redirect/`.
-* Under `Allowed Logout URLs`, add `https://app.<your-astronomer-base-domain>/logout`.
-* Under `Allowed Origins (CORS)`, add `https://*.<your-astronomer-base-domain>`.
+To let users authenticate to Astronomer with a local username and password, follow the steps below.
 
-**Create Auth0 API:**
+1. Enable Local Auth in your `config.yaml` file:
```
Comment (lzdanski) [inline software/integrate-auth-system.md:268]:
```suggestion
1. Enable `local auth` in your `config.yaml` file:
```

File: software/integrate-auth-system.md [hunk 5]
```diff
@@ -591,3 +603,7 @@ This setup is primarily used for encrypting the required secrets for [configurin
     ```
 
 4. Save and push your changes. See [Apply a config change](apply-platform-config.md).
+
+## Next steps
+
+After
```
Comment (lzdanski) [inline software/integrate-auth-system.md:609]: Missing some content here!

**Comment by lzdanski** [comment]
First pass review - just some copy and consistency errors. It looks like some content didn't make it to the `next steps` on one of the pages.

File: software/integrate-auth-system.md [hunk 6]
```diff
@@ -225,76 +167,144 @@ astronomer:
...
+</TabItem>
+<TabItem value="autho">
+
+If you manage your identity provider through Auth0, follow these steps to configure the identity provider for Astro.
+
+#### Step 1: Create an Auth0 tenant domain
+
+Follow the Auth0 documentation to [create a tenant](https://auth0.com/docs/get-started/auth0-overview/create-tenants). You can use the default domain name or your own unique `tenant-name`. Your full tenant domain will look something like `astronomer.auth0.com`.
+
+> **Note:** Your full tenant domain may differ if you've created it outside of the United States.
```
Comment (jwitz) [inline software/integrate-auth-system.md:183]:
```suggestion
:::info

Your full tenant domain may differ if you've created it outside of the United States.

:::
```

File: software/integrate-auth-system.md [hunk 7]
```diff
@@ -225,76 +167,144 @@ astronomer:
...
+        openidConnect:
+          auth0:
+            enabled: true
+            clientId: "<default-app-client-id>"
+            discoveryUrl: https://<tenant-name>.auth0.com
+```
 Then, push the configuration change to your platform as described in [Apply a config change](apply-platform-config.md).
 
-> **Note:** `okta-base-domain` will be different from the basedomain of your Software installation. You can read [Okta's docs on finding your domain](https://developer.okta.com/docs/api/getting_started/finding_your_domain/) if you are unsure what this value should be.
+> **Note:** You can find your `clientID` value at `https://manage.auth0.com/dashboard/us/<tenant-name>/applications` listed next to 'Default App'.
```
[the remark this answers is not in the document]
Comment (jwitz) [reply software/integrate-auth-system.md:221]: This is leftover formatting from the olden days that I forgot to fix here. Any time we see it, we should remove it

Comment (jwitz) [inline software/integrate-auth-system.md:221]:
```suggestion
:::info 
You can find your `clientID` value at `https://manage.auth0.com/dashboard/us/<tenant-name>/applications` listed next to 'Default App'.
:::
```

File: software/integrate-auth-system.md (hunk 5 above)
Comment (jwitz) [inline software/integrate-auth-system.md:609]:
```suggestion
```

File: software/integrate-auth-system.md [hunk 8]
```diff
@@ -1,113 +1,56 @@
...
-Your Cognito pool ID can be found in the `General settings` tab of the Cognito portal. Your client ID is found in the `App clients` tab.
-
-Once you've saved your `config.yaml` file with these values, push it to your platform. See [Apply a config change](apply-platform-config.md).
-
-### Microsoft Entra ID
-
-Follow these steps to configure Microsoft Entra ID as your OIDC provider. 
+<Tabs
+    groupId="setup"
+    defaultValue="azuread"
```
Comment (jwitz) [inline software/integrate-auth-system.md:42]:
```suggestion
    defaultValue="entraid"
```

**Comment by jwitz** [comment]
@lzdanski I think that "Next steps" was leftover from a draft, removed that section and accepted all other suggestions!

**Pull request merged**